curvefs/src/metaserver/metastore.cpp

```diff
@@ -737,12 +737,14 @@ MetaStatusCode MetaStoreImpl::GetOrModifyS3ChunkInfo(
 }
 
 void MetaStoreImpl::PrepareStreamBuffer(butil::IOBuf* buffer,
-                                        uint64_t chunkIndex,
-                                        const std::string& value) {
+                             uint64_t chunkIndex,
+                             const char* data, size_t data_size) {
```

Comment:
ditto, if `absl::string_view` works, then we can use it.

Reply:
I have the same suggestion: https://github.com/abseil/abseil-cpp/blob/master/absl/strings/string_view.h

```
// A `string_view` provides a lightweight view into the string data provided by
// a `std::string`, double-quoted string literal, character array, or even
// another `string_view`. A `string_view` does *not* own the string to which it
// points, and that data cannot be modified through the view.
```

Reply:
> I have the same suggestion: https://github.com/abseil/abseil-cpp/blob/master/absl/strings/string_view.h
> 
> ```
> // A `string_view` provides a lightweight view into the string data provided by
> // a `std::string`, double-quoted string literal, character array, or even
> // another `string_view`. A `string_view` does *not* own the string to which it
> // points, and that data cannot be modified through the view.
> ```

string_view is definitely neater solution here, but I still have the same concern as I replayed above, using string_view as returned value of Value() is not safe for other derived iterators. As stated in string_view's comments:

```
// Due to lifetime issues, a `string_view` is sometimes a poor choice for a
// return value and usually a poor choice for a data member. If you do use a
// `string_view` this way, it is your responsibility to ensure that the object
// pointed to by the `string_view` outlives the `string_view`.
```

For RocksDBStorageIterator, since RocksDB has already guaranteed that its key & value will live as long as the iterator's life time, returning string_view is ok, but I don't see other iterators have such guarantee.